src/meta/processors/admin/BalanceTask.h

```diff
@@ -73,7 +75,17 @@ class BalanceTask {
 
     void rollback();
 
-    Result result() const {
+    void markStopIfNotStarted() {
+        std::lock_guard<std::mutex> lg(lock_);
+        if (status_ == Status::START) {
+            LOG(INFO) << "mark stopped of task " << partId_;
+            ret_ = Result::STOPPED;
+            saveInStore();
+        }
+    }
+
+    Result result() {
+        std::lock_guard<std::mutex> lg(lock_);
```

Comment:
is here really need lock?

Reply:
Not really, it is used to show the status of balance task. It is ok if it is outdated.